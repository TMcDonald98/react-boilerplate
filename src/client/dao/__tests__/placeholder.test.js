import {
  getPlaceholder,
  postPlaceholder,
  putPlaceholder,
  deletePlaceholder
} from '../placeholder';

const serviceHosts = {
  serviceHost1: 'host'
};

describe('dao/placeholder', () => {
  let request;

  beforeEach(() => {
    request = jest.fn(arg => arg);
  });

  describe('getPlaceholder', () => {
    it('should return the correct dao object', () => {
      const result = getPlaceholder(
        {
          request,
          serviceHosts
        },
        10
      );

      expect(result).toMatchSnapshot('getPlaceholder');
    });
  });

  describe('putPlaceholder', () => {
    it('should return the correct dao object', () => {
      const result = putPlaceholder(
        {
          request,
          serviceHosts
        },
        'test'
      );

      expect(result).toMatchSnapshot('putPlaceholder');
    });
  });

  describe('postPlaceholder', () => {
    it('should return the correct dao object', () => {
      const result = postPlaceholder(
        {
          request,
          serviceHosts
        },
        'test'
      );

      expect(result).toMatchSnapshot('postPlaceholder');
    });
  });

  describe('deletePlaceholder', () => {
    it('should return the correct dao object', () => {
      const result = deletePlaceholder(
        {
          request,
          serviceHosts
        },
        'test'
      );

      expect(result).toMatchSnapshot('deletePlaceholder');
    });
  });
});
