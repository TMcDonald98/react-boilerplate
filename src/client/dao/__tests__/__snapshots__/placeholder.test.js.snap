// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dao/placeholder deletePlaceholder should return the correct dao object: deletePlaceholder 1`] = `
Object {
  "method": "DELETE",
  "params": Object {
    "placeholderId": "test",
  },
  "path": "/v1/placeholder",
  "service": "host",
}
`;

exports[`dao/placeholder getPlaceholder should return the correct dao object: getPlaceholder 1`] = `
Object {
  "auth": false,
  "method": "GET",
  "params": Object {
    "limit": 10,
  },
  "path": "/v1/placeholder",
  "service": "host",
}
`;

exports[`dao/placeholder postPlaceholder should return the correct dao object: postPlaceholder 1`] = `
Object {
  "data": "test",
  "method": "POST",
  "path": "/v1/placeholder",
  "service": "host",
}
`;

exports[`dao/placeholder putPlaceholder should return the correct dao object: putPlaceholder 1`] = `
Object {
  "data": "test",
  "method": "PUT",
  "path": "/v1/placeholder",
  "service": "host",
}
`;
